pkg/kv/bulk/kv_buf_test.go

```diff
@@ -51,19 +55,40 @@ func TestKvBuf(t *testing.T) {
 
 	src, totalSize := makeTestData(50000)
 
+	ctx := context.Background()
+	none := mon.NewMonitorWithLimit("none", mon.MemoryResource, 0, nil, nil, 0, 0, nil).MakeBoundAccount()
+	lots := mon.NewUnlimitedMonitor(ctx, "lots", mon.MemoryResource, nil, nil, 0, nil).MakeBoundAccount()
+
 	// Write everything to our buf.
 	b := kvBuf{}
 	for i := range src {
+		size := sz(len(src[i].key) + len(src[i].value))
```

Comment:
I'd prefer if we had a dedicated unit test for `fits` that gives us as much coverage of the various branches in that method as possible. For egs: the logic around not wasting an alloc on the slab if the entries cannot grow to a size that points to the slab. Maybe these cases already tickle that. Not a blocking comment, but it looks like its relatively lightweight to setup?

Reply:
Good point! I ran a coverage report and this test is actually covering almost all of `fits` as-is, with the only cases uncovered being the maxing out of grow sizes, just since the test doesn't get up to that many megabytes:
<img width="1486" alt="Screen Shot 2022-03-19 at 7 26 29 PM" src="https://user-images.githubusercontent.com/15615/159141799-b4218a58-41bc-4923-9bc1-ee3666060e12.png">
.

So we do at least know the code is _running_ and that when doing so, it isn't violating the "correctness" promise that after getting a `true` from a `fits` call, a subsequent `append` call never grows `memSize` since the test does check that. Where we might have more of a gap though is that we don't test that we did "good" job of splitting the allocation between the slab and entries. I wonder if a test can check that, of maybe a benchmark. I'll ponder a little.

Reply:
Thanks, the coverage gives me more confidence!